src/wp-includes/html-api/class-wp-html-tag-processor.php

```diff
@@ -2741,19 +3200,25 @@ public function has_self_closing_flag(): bool {
 	 * @return bool Whether the current tag is a tag closer.
 	 */
 	public function is_tag_closer(): bool {
-		return (
-			self::STATE_MATCHED_TAG === $this->parser_state &&
-			$this->is_closing_tag &&
+		if ( self::STATE_MATCHED_TAG !== $this->parser_state ) {
+			return false;
+		}
 
-			/*
-			 * The BR tag can only exist as an opening tag. If something like `</br>`
-			 * appears then the HTML parser will treat it as an opening tag with no
-			 * attributes. The BR tag is unique in this way.
-			 *
-			 * @see https://html.spec.whatwg.org/#parsing-main-inbody
-			 */
-			'BR' !== $this->get_tag()
-		);
+		switch ( $this->get_namespace() ) {
+			case 'html':
+				/*
+				 * The BR tag can only exist as an opening tag. If something like `</br>`
+				 * appears then the HTML parser will treat it as an opening tag with no
+				 * attributes. The BR tag is unique in this way.
+				 *
+				 * @see https://html.spec.whatwg.org/#parsing-main-inbody
+				 */
+				return $this->is_closing_tag && 'BR' !== $this->get_tag();
```

Comment:
This can probably be reverted because BR tags will close out of foreign content.

Reply:
Reverted in f4373ff